projects/klippa/ngx-enhancy-forms/src/lib/form/form-submit-button/form-submit-button.component.ts

```diff
@@ -16,44 +16,36 @@ export type SubmitButtonVariant = Extract<ButtonVariant,
 	styleUrls: ['./form-submit-button.component.scss'],
 })
 export class FormSubmitButtonComponent {
+	private parentForm = inject(FormComponent, {host: true, optional: true});
+	private handleError = inject(KLP_FORM_ERROR_HANDLER, {optional: true}) ?? DefaultErrorHandler;
+
 	@Input() public isLoading = false;
-	@Input() fullWidth = false;
-	@Input() variant: SubmitButtonVariant = 'greenFilled';
+	@Input() @HostBinding('class._fullWidth') public fullWidth = false;
+	@Input() public variant: SubmitButtonVariant = 'greenFilled';
+	@Input() public submitCallback: (renderedAndEnabledValues: object, renderedButDisabledValues: object) => Promise<any>;
 	@Input() public before: () => Promise<any> = () => Promise.resolve();
 	@Input() public after: () => Promise<any> = () => Promise.resolve();
-	@Input() public submitCallback: (renderedAndEnabledValues: object, renderedButDisabledValues: object) => Promise<any>;
 
-	@HostBinding('class._fullWidth') get _() {
-		return this.fullWidth;
+	private renderError = (e: FormValidationError) => {
+		this.parentForm.formGroup.get(e.path)?.setErrors({ message: { value: e.message }});
 	}
 
-	constructor(@Host() @Optional() private parentForm: FormComponent) {}
-
 	async submitForm(): Promise<void> {
+		await this.before().catch(() => null);
+
 		try {
-			await this.before();
+			const [renderedAndEnabledValues, renderedValues] = await this.parentForm.trySubmit();
+			this.isLoading = true;
+			await this.submitCallback(renderedAndEnabledValues, renderedValues)
+				.finally(() => this.isLoading = false);
 		} catch (e) {
+			if (e === invalidFieldsSymbol) {
+				return;
```

Comment:
I prefer to leave the comment explaining why the error was swallowed. What is the reason for removing that?

Reply:
I also prefer those comments! I must have removed it by mistake.